Fixes and additions.

d8ahazard opened the pull request:
Set correct device ID for connect, add solo definitions.
Remove unnecessary passing of "offset" value around code.
Decode device ID on message receipt and dynamically load proper byte offsets to decode state messages.
Add initial calls for color/saturation/brightness setters getters.

d8ahazard committed:
Fix Connect device ID, add solo Device ID

Connect ID is 4, not 6. My mistake.

Add device ID of 7 for DS solo.

Signed-off-by: d8ahazard <d8ahazard@gmail.com>

d8ahazard committed:
Add handler and thing definition for DS solo

Literally just copypaste the DS HD definitions and change the ID.

Signed-off-by: d8ahazard <d8ahazard@gmail.com>

d8ahazard committed:
Fix whitespace issues

Signed-off-by: d8ahazard <d8ahazard@gmail.com>

d8ahazard committed:
We really, really don't need this offset stuff.

Remove all references/calls/passes to "off". Offset is always zero, treat it as such.

Signed-off-by: d8ahazard <d8ahazard@gmail.com>

d8ahazard committed:
Dynamically load byte offsets based on device type

Decode our device type when receiving a message and store it in the super. This way, when we get state messages, we know the correct offsets from which to grab data.

Signed-off-by: d8ahazard <d8ahazard@gmail.com>

markus7017 commented on bundles/org.openhab.binding.dreamscreen/src/main/java/org/openhab/binding/dreamscreen/internal/message/DreamScreenMessage.java:
```diff
@@ -30,56 +31,56 @@
-
-        if (len > 6 && data[off] == (byte) 0xFC) {
-            msgLen = data[off + 1] & 0xFF;
+        if (len > 6 && data[0] == (byte) 0xFC) {
```
please add a constant for 0xFC

d8ahazard replied:
Done

markus7017 commented on bundles/org.openhab.binding.dreamscreen/src/main/java/org/openhab/binding/dreamscreen/internal/message/RefreshMessage.java:
```diff
@@ -28,46 +28,68 @@
-    }
+    // Set integers so we know which bytes to pull based on device
+
+    private int groupInt = 32;
```
make them static final int and constant names in upper case, (required to pass the review; all of them) e.g.
private static final int OFF_GROUP_ID = 32;

markus7017 commented on bundles/org.openhab.binding.dreamscreen/src/main/java/org/openhab/binding/dreamscreen/internal/message/RefreshMessage.java:
```diff
@@ -81,4 +103,36 @@ public DatagramPacket writePacket(InetAddress address, int port) {
+
+    private void refreshType() {
+        switch (super.deviceType) {
+            case 1:
```
please use PRODUCT_ID_xxx instead of hard coded device IDs

markus7017 commented on bundles/org.openhab.binding.dreamscreen/src/main/java/org/openhab/binding/dreamscreen/internal/message/RefreshTvMessage.java:
```diff
@@ -26,14 +26,14 @@
+        if (RefreshMessage.matches(data)) {
+            final int msgLen = data[1] & 0xFF;
+            final byte productId = data[msgLen];
             return productId == DreamScreenHdHandler.PRODUCT_ID || productId == DreamScreen4kHandler.PRODUCT_ID;
```
dod we need to add additional model IDs?

d8ahazard replied:
I think we do. Is this only supposed to handle DS devices, or all devices?

d8ahazard replied:
Added handler for solo (for now).

markus7017 commented on bundles/org.openhab.binding.dreamscreen/src/main/resources/ESH-INF/config/config.xml:
```diff
@@ -10,10 +10,10 @@
-            <label>Group Id</label>
-            <description>Logical group Id to control multiple devices at once</description>
-        </parameter>
+		<parameter name="groupId" type="text">
```
please run "mvn spotless:apply" to fix formatting, this is required to pass the review

markus7017 commented on bundles/org.openhab.binding.dreamscreen/src/main/resources/ESH-INF/thing/connect.xml:
```diff
@@ -1,24 +1,24 @@
-    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
-    xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
please run "mvn spotless:apply" to fix formatting, this is required to pass the review

markus7017 reviewed (changes requested):
please run "mvn spotless:apply" to fix formatting, this is required to pass the review
I'm not sure if we could turn the offsets into constants (saw that they are somehow device specific)
If the offset stuff is not required we could remove it

d8ahazard committed:
Add saturation, brightness, channels and methods.

Signed-off-by: d8ahazard <d8ahazard@gmail.com>

d8ahazard committed:
Use product_id constants instead of hardcoding

Signed-off-by: d8ahazard <d8ahazard@gmail.com>

d8ahazard committed:
Refactor offset names, change access to static

Can't make them final as we may need to change them. Could probably be loaded better, this was just the first thing I came up with.

Signed-off-by: d8ahazard <d8ahazard@gmail.com>

d8ahazard committed:
Maven/spotless cleanup

Signed-off-by: d8ahazard <d8ahazard@gmail.com>

d8ahazard committed:
Add constant for magic byte

Signed-off-by: d8ahazard <d8ahazard@gmail.com>

d8ahazard committed:
Add handler for solo, not sure if we need sidekick or connect in there as well...

Signed-off-by: d8ahazard <d8ahazard@gmail.com>
